src/maintenance.js

```diff
@@ -133,3 +134,167 @@ module.exports.rotate = async function({Namespace, now, cfg, rabbitManager}) {
   });
   return count;
 };
+
+function decideState(queue, cfg) {
+  let currentState = 'normal';
+  let subject = `${queue.name} has returned to a safe state`;
+  let content = `
+The number of messages queued in \`${queue.name}\` is now below ${cfg.alertThreshold}.
+No further action is necessary on your part, although you may want to investigate
+why this happened in the first place.`;
+
+  if (queue.messages > cfg.deleteThreshold) {
+    currentState = 'danger';
+    subject = `${queue.name} has been deleted!`;
+    content = `
+The number of messages queued in \`${queue.name}\` exceeded ${cfg.deleteThreshold}.
+At the time of deletion, there were ${queue.messages} messages in the queue.`;
+  } else if (queue.messages > cfg.alertThreshold) {
+    currentState = 'warning';
+    subject = `${queue.name} has is in danget of being deleted!`;
+    content = `
+The number of messages queued in \`${queue.name}\` is now above ${cfg.alertThreshold}.
+Currently there are ${queue.messages} messages in the queue. If this number goes
+above ${cfg.deleteThreshold}, the queue will be deleted and all of the messages
+will be lost.
+
+A common cause of this situation is that your service has crashed.`;
+  }
+  return {currentState, subject, content};
+}
+
+async function needMessage(name, currentState, RabbitQueue) {
+  let rq;
+  let sendMessage = false;
+  try {
+    rq = await RabbitQueue.load({name});
+
+    rq.modify((entity) => {
+      if (entity.state !== currentState) {
+        entity.state = currentState;
+        entity.updated = new Date();
+        sendMessage = true;
+      }
+    });
+  } catch (e) {
+    if (e.code !== 'ResourceNotFound') {
+      throw e;
+    }
+    rq = await RabbitQueue.create({
+      name,
+      state: currentState,
+      updated: new Date(),
+    });
+    if (currentState !== 'normal') {
+      sendMessage = true;
+    }
+  }
+  return sendMessage;
+}
+
+async function handleQueues({cfg, prefix, manager, Namespace, RabbitQueue, notify}) {
+  let debug = Debug('maintenance.handle-queues');
+  await Promise.map(await manager.queues(), async queue => {
+    if (!queue.name.startsWith(cfg.queuePrefix + prefix)) {
+      return; // Note: This is very important to avoid stepping on pulseguardian's toes
+    }
+
+    let {currentState, subject, content} = decideState(queue, cfg);
+    let namespace = queue.name.split('/')[1];
```

Comment:
I think this is assuming `cfg.monitor.queuePrefix` doesn't have slashes.

In other parts of the code, I got tricky and used the permission regexps (`"^(queue/{{namespace}}/.*|exchange/{{namespace}}/.*)"`) for that.  You *might* be able to do the same (replacing `{{namespace}}` with a capture group)?  Otherwise, maybe a new config `cfg.app.queueNamePrefix` set to `queue/` would be the thing to use.